subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -2290,21 +2301,25 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 						  0, 0, ticker_update_op_cb);
 
 #if defined(CONFIG_BT_CTLR_ADV_EXT)
-		adv->event_counter += (lazy + 1);
-
-		if (adv->ticks_remain_duration) {
-			uint32_t ticks_interval =
-				HAL_TICKER_US_TO_TICKS((uint64_t)adv->interval *
-						       ADV_INT_UNIT_US);
-			uint32_t ticks_elapsed = ticks_interval * (lazy + 1) +
-						 ticks_drift;
-
-			if (adv->ticks_remain_duration > ticks_elapsed) {
-				adv->ticks_remain_duration -= ticks_elapsed;
+		if (adv->remain_duration_us && adv->event_counter > 0) {
+#if defined(CONFIG_BT_CTLR_JIT_SCHEDULING)
+			/* ticks_drift is always 0 with JIT scheduling, populate manually */
+			ticks_drift = adv->delay_at_expire;
+#endif /* CONFIG_BT_CTLR_JIT_SCHEDULING */
+			uint32_t interval_us = (uint64_t)adv->interval * ADV_INT_UNIT_US;
+			uint32_t elapsed_us = interval_us * (lazy + 1) +
+						 HAL_TICKER_TICKS_TO_US(ticks_drift);
+
+			/* End advertising if the added random delay pushes us beyond the limit */
+			if (adv->remain_duration_us > elapsed_us + interval_us +
+						      HAL_TICKER_TICKS_TO_US(random_delay)) {
```

Comment:
`elapsed_us` consists of current event's `ticks_drift` plus no. of skipped intervals (represented by `lazy`) before the current event plus the next interval (`+ 1`).

I agree with you for adding `random_delay`  which is guess here in the `ticker_cb`  as the ticks slow window feature would nudge it when there are overlaps, and hence in the next interval's `ticker_cb` we will use the actual `ticks_drift` to reduce the `remain_duration_us`.

```suggestion
			if (adv->remain_duration_us > elapsed_us +
						      HAL_TICKER_TICKS_TO_US(random_delay)) {
```